src/Automation/Automation/Model/UpdateManagement/SoftwareUpdateConfiguration.cs

```diff
@@ -100,19 +100,21 @@ internal SoftwareUpdateConfiguration(string resourceGroupName, string automation
                 foreach (var query in suc.UpdateConfiguration.Targets.AzureQueries)
                 {
                     var tags = new Dictionary<string, List<string>>();
-                    foreach (var tag in query.TagSettings.Tags)
+                    if(query.TagSettings.Tags != null)
```

Comment:
@jemex do we want to also include a null check on `query` and `query.TagSettings` here?